src/test/rotary_test.cpp

```diff
@@ -17,6 +17,13 @@ TEST(Rotary, basicOverflow) {
     EXPECT_EQ(rot.filter(-2), -2);
 }
 
+TEST(Rotary, noQuantization) {
+    Rotary rot(2);
+    EXPECT_EQ(rot.filter(0.25), 0.125);
+    EXPECT_EQ(rot.filter(0.0), 0.125);
+    EXPECT_EQ(rot.filter(0.0), 0.0);
```

Comment:
Small note: Yes, we do precise comparisons on floats here, which is generally discouraged, but powers of two are well-representable in IEEE 754, so we should be fine.

Reply:
or just use `EXPECT_DOUBLE_EQ` just to be sure. The ringbuffer is not large enough to suffer from large inaccuracies.